cmake/app/boilerplate.cmake

```diff
@@ -550,20 +550,24 @@ include(${BOARD_DIR}/board.cmake OPTIONAL)
 # The Qemu supported ethernet driver should define CONFIG_ETH_NIC_MODEL
 # string that tells what nic model Qemu should use.
 if(CONFIG_QEMU_TARGET)
-  if(CONFIG_NET_QEMU_ETHERNET)
-    if(CONFIG_ETH_NIC_MODEL)
+  if(CONFIG_ETH_NIC_MODEL)
+    if(CONFIG_NET_QEMU_ETHERNET)
```

Comment:
This still does not seem right ... It should be something like:
```
if(CONFIG_NET_QEMU_ETHERNET OR CONFIG_NET_QEMU_USER)
  if(CONFIG_ETH_NIC_MODEL)
    if(CONFIG_NET_QEMU_ETHERNET)
      ...
    elseif(CONFIG_NET_QEMU_USER)
      ...
    endif()
  else()
    (NO QEMU ETH DRIVER CONFIGURED!)
  endif()
endif()
```

or to reduce nesting:
```
if ((CONFIG_NET_QEMU_ETHERNET OR CONFIG_NET_QEMU_USER) AND NOT CONFIG_ETH_NIC_MODEL)
  (NO QEMU ETH DRIVER CONFIGURED!)
endif()
```

Reply:
> if (CONFIG_NET_QEMU_ETHERNET OR CONFIG_NET_QEMU_USER)

I can add this check but I'm not sure about the value it would add. In the current implementation, we check if "NIC_MODEL" is defined and set flags based on the "NET_QEMU_X" option. If the "NIC_MODEL" is not set, we spit out an error. 

In the proposed flow, we would be doing the same expect for an additional check.. The result wouldn't change from what I see.. May be I'm missing something ?

Reply:
Okay, I see now.. Will fix..